### PR TITLE
Mirror did doc semantics

### PR DESCRIPTION
**Pull request by dholms**
This updates our data model so that it more closely mirrors did doc semantics.

Namely:
- updating "handle" to "alsoKnownAs"
- updating "services" to a map
- updating "signingKey" to "verificationMethods"

We have room to add in additional did doc semantics - such as verification relationships - in future work.

This also introduces some operation formatters on both the client & the library for easily working with atproto formatted DIDs

### CODE REVIEW

File: packages/lib/src/operations.ts [hunk 1]
```diff
@@ -20,6 +20,164 @@ export const didForCreateOp = async (op: t.CompatibleOp, truncate = 24) => {
...
+  signer: Keypair,
+  endpoint: string,
+): Promise<t.Operation> => {
+  const formatted = ensureHttpPrefix(endpoint)
+  return createUpdateOp(lastOp, signer, (normalized) => {
+    return {
+      ...normalized,
+      services: {
+        ...normalized.services,
+        atproto_pds: {
```
Comment (devinivy) [inline packages/lib/src/operations.ts:143]: Just curious, where does the snake-casing convention stem from in this context?

**Review by devinivy — APPROVED** [review]
This feels like it was definitely a good call.

File: packages/lib/tests/document.test.ts [hunk 2]
```diff
@@ -5,100 +5,77 @@ import * as t from '../src/types'
...
+          endpoint: atpPds,
+        },
+        other: {
+          type: 'SomeService',
+          endpoint: otherService,
+        },
       },
     }
     const doc = await document.formatDidDoc(data)
+    // only epxected keys
```
Comment (devinivy) [inline packages/lib/tests/document.test.ts:35]:
```suggestion
    // only expected keys
```

File: packages/lib/tests/document.test.ts [hunk 3]
```diff
@@ -5,100 +5,77 @@ import * as t from '../src/types'
 
 describe('document', () => {
```
Comment (devinivy) [inline packages/lib/tests/document.test.ts:6]: Is there any blessed library out there for validating a did doc?  I poked around but didn't turn one up.  Just could see it being useful for these tests, to ensure some level of compatibility.

File: packages/lib/src/operations.ts (hunk 1 above)
Comment (dholms) [reply packages/lib/src/operations.ts:143]: for two reasons:
- the keys of this map become the `id` in the context of the did doc, which i believe are meant to be case-insensitive: https://www.w3.org/TR/did-core/#services
- but you can't write record keys in kebab-case (at least nicely, you can wrap them in quotes i guess) so I used snake_case instead

Comment (devinivy) [reply packages/lib/src/operations.ts:143]: Right on, thanks!

File: packages/lib/tests/document.test.ts (hunk 3 above)
Comment (dholms) [reply packages/lib/tests/document.test.ts:6]: actually this seems like the closest thing: https://www.npmjs.com/package/@did-core/did-ld-json

i'll look at hooking it up 🧐

Comment (dholms) [reply packages/lib/tests/document.test.ts:6]: eh this is not helpful at all & I'm like 98% sure I have the formatting correctly. 

fortunately, changing our did doc representation is an easy change to maek

Comment (devinivy) [reply packages/lib/tests/document.test.ts:6]: Sweet, sounds good!

**Pull request merged**